<launch>
    <!-- add vehicle dimensions to parameter server (values below are approximate and relative to lidar) -->
    <param name="/vehicles/ego" value="touareg"/>
    <param name="/vehicles/touareg/geometric/height_ref_to_maximum" value="1.9"/>
    <param name="/vehicles/touareg/geometric/height_ref_to_ground" value="-0.64"/>
    <param name="/vehicles/touareg/geometric/length_ref_to_front_end" value="2.3535"/>
    <param name="/vehicles/touareg/geometric/length_ref_to_rear_end" value="-2.4005"/>
    <param name="/vehicles/touareg/geometric/width_ref_to_left_mirror" value="1.1085"/>
    <param name="/vehicles/touareg/geometric/width_ref_to_right_mirror" value="-1.1085"/>

    <!-- visualize dummy car in Rviz (has not the correct dimensions!) -->
    <param name="robot_description" command="$(find xacro)/xacro $(find continuous_clustering)/urdf/dummy_car.urdf"/>

    <!-- uncomment the following if you want to have the regular 360Â° PointCloud2 for the Ouster sensor -->

    <!-- lidar ouster left -->
    <!--<group>
        <remap from="lidar_packets" to="/bus/os32_left/lidar_packets"/>
        <remap from="imu_packets" to="/bus/os32_left/imu_packets"/>
        <remap from="metadata" to="/bus/os32_left/metadata"/>
        <remap from="get_metadata" to="/bus/os32_left/get_metadata"/>

        <include file="$(find ouster_ros)/launch/replay.launch">
            <arg name="ouster_ns" value="sensor/lidar/os32_left"/>
            <arg name="metadata" value="$(find continuous_clustering)/calibrations/touareg_os32_left.json"/>
            <arg name="viz" value="false"/>
            <arg name="timestamp_mode" default="TIME_FROM_ROS_TIME"/>
            <arg name="tf_prefix" default="sensor/lidar/os32_left"/>
        </include>
    </group>-->

    <!-- lidar ouster right -->
    <!--<group>
        <remap from="lidar_packets" to="/bus/os32_right/lidar_packets"/>
        <remap from="imu_packets" to="/bus/os32_right/imu_packets"/>
        <remap from="metadata" to="/bus/os32_right/metadata"/>
        <remap from="get_metadata" to="/bus/os32_right/get_metadata"/>

        <include file="$(find ouster_ros)/launch/replay.launch">
            <arg name="ouster_ns" value="sensor/lidar/os32_right"/>
            <arg name="metadata" value="$(find continuous_clustering)/calibrations/touareg_os32_right.json"/>
            <arg name="viz" value="false"/>
            <arg name="timestamp_mode" default="TIME_FROM_ROS_TIME"/>
            <arg name="tf_prefix" default="sensor/lidar/os32_right"/>
        </include>
    </group>-->
</launch>